### PR TITLE
Github Actions (CI) -- Add concurrency to content-release based off environment

### PR DESCRIPTION
**Pull request by oseasmoran73**
## Description

As discussed in [slack](https://dsva.slack.com/archives/C01512KN35G/p1635533884032500?thread_ts=1635529221.019000&cid=C01512KN35G) thread, CR workflow now queued based off of their environment

## Testing done

N/A, will need to merge first and test dispatch. Important thing to notice is 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .github/workflows/content-release.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ on:
   #   - cron: "0 13-16,20-21 * * 1-5"
   #   - cron: "45 17 * * 1-5"
 
+concurrency:
+  group: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.deploy_environment || 'prod' }}
```
Comment (oseasmoran73) [inline .github/workflows/content-release.yml:19]: Translated to:

workflow_dispatch ? deploy_environment : prod

Comment (alexandec) [reply .github/workflows/content-release.yml:19]: Is there another way to write this line? Your translation is helpful, but without that it's hard to parse, or easy to misinterpret

**Review by alexandec — APPROVED** [review]

**Comment by alexandec** [comment]
Would be good to add a link to the thread, and/or a bit more info about this change

File: .github/workflows/content-release.yml (hunk 1 above)
Comment (oseasmoran73) [reply .github/workflows/content-release.yml:19]: Unfortunately there is no way. ideally I would like to have it in the syntax provided above, but GHA does not allow that type of format 😭

**Comment by oseasmoran73** [comment]
After talking with CMS team, we want to add concurrency to content-release. Since dev and stage will be enabled with prod, they will be grouped with their respect environment.

The only flows that should be cancelled if a new one gets kicked off are for dev & staging. Prod should never cancel, but rather queue

File: .github/workflows/content-release.yml [hunk 2]
```diff
@@ -15,6 +15,10 @@ on:
...
+  cancel-in-progress: ${{ github.event.inputs.deploy_environment != 'prod' && true || false }}
```
Comment (oseasmoran73) [inline .github/workflows/content-release.yml:20]: Translated to:

deploy_environment != prod ? true || false

If the environment is dev/stage, cancel all flows associated with group-name. Otherwise, if prod, queue and operate accordingly (much like Jenkins)
